homeassistant/components/xiaomi_miio/fan.py

```diff
@@ -905,10 +904,10 @@ def __init__(self, name, device, entry, unique_id, allowed_failures=0):
             self._device_features = FEATURE_FLAGS_AIRPURIFIER
             self._available_attributes = AVAILABLE_ATTRIBUTES_AIRPURIFIER
             self._preset_modes = PRESET_MODES_AIRPURIFIER
-            self._supported_features = SUPPORT_SET_SPEED | SUPPORT_PRESET_MODE
-            self._speed_count = 4
+            self._supported_features = SUPPORT_PRESET_MODE
+            self._speed_count = 1
```

Comment:
Looks otherwise good to me, but should this be a zero for an empty speed list (that are going to be removed soonish)?

Reply:
I kept it in line with the reset of the code. If speed is not enabled it has no effect. The default is 100.
https://developers.home-assistant.io/docs/core/entity/fan/
Setting it to 0 could make the `speed_step calculation` invalid.